firmware/library/L2_HAL/temperature.hpp

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include <cstdint>
+#include <cstdio>
+#include "L1_Drivers/i2c.hpp"
```

Comment:
You aren't using any of these includes. Including them will only slow down compilation time. Do not include what are not using. Always include what you are using.

With my comment below, you should keep <cstdint> though.